kv/task.go

```diff
@@ -951,6 +951,21 @@ func (s *Service) findRuns(ctx context.Context, tx Tx, filter influxdb.RunFilter
 	if filter.Limit < 0 || filter.Limit > influxdb.TaskMaxPageSize {
 		return nil, 0, influxdb.ErrOutOfBoundsLimit
 	}
+	parsedFilterAfterTime := time.Time{}
+	parsedFilterBeforeTime := time.Now().UTC()
+	var err error
+	if len(filter.AfterTime) > 0 {
+		parsedFilterAfterTime, err = time.Parse(time.RFC3339, filter.AfterTime)
+		if err != nil {
+			return nil, 0, err
+		}
+	}
+	if filter.BeforeTime != "" {
+		parsedFilterBeforeTime, err = time.Parse(time.RFC3339, filter.BeforeTime)
+		if err != nil {
+			return nil, 0, err
+		}
+	}
```

Comment:
nit: this logic could someday be moved into a function to dry up the code (it's reused in `task/backend/analytical_storage.go:157`)

Reply:
Unsure how I would go about doing that as they are in different packages/folders. Would I just make two of the same helper functions in each file?

Reply:
i'm not sure the proper location (maybe in this same file in the `kit` package that you would then import in `task/backend/analytical_storage`), but defining the logic once to a central location and importing/calling it in others helps when that logic needs updated. no need to worry about it this go